src/main/java/com/puppycrawl/tools/checkstyle/api/TreeStringPrinter.java

```diff
@@ -0,0 +1,41 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Interface for printing ASTs to String.
+ */
+@FunctionalInterface
+public interface TreeStringPrinter {
```

Comment:
I dont like this design, to be honest.

method name like `printFileAst` assumes that 
1. there is no return, similar to `System.out.println()` and similar methods.
2. Result is the AST, which is weird since return type is String, which means any text.

Plus, name of the interface is confusing, since even existing implementations dont do any printing of the AST, they just convert it into string. Printing is performed outside, in Main.java line 369.

Therefore I suggest to rename it to something like FileConverter (or something like this) with meaningful method name.

Reply:
@nrmancuso @strkkk @pbludov 
Is there any agreement we can make on design before I continue? @nrmancuso Was mentioning extension over something being built into the project. For @strkkk 's comment, everything was originally `printFileAst` so thats why I went with this, but if we head down extension, then naming becomes not an issue as 3rd parties will be the ones to name it themselves.

Reply:
Can we make https://github.com/checkstyle/checkstyle/blob/a8f065738de6c4ff917a2ee57521f6cf8718f2ba/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L629 a public class to allow extension of options? There is bound to be more things that some third party may need to do with the CLI, and this way they would not be blocked by limitations of checksytle CLI.